GVFS/GVFS.Platform.Mac/ProcessLauncher.cs

```diff
@@ -49,36 +50,40 @@ public static void StartBackgroundProcess(string programName, string[] args)
         }
 
         private static unsafe void RunChildProcess(
+            ITracer tracer,
             string programName, 
             byte** argvPtr = null,
             byte** envpPtr = null)
         {
-            // TODO(Mac): Issue #583, log errors here to a log file
-
             int fdin = Open("/dev/null", (int)NetCoreMethods.OpenFlags.O_RDONLY);
+            if (fdin == -1)
+            {
+                LogErrorAndExit(tracer, "Unable to open file descriptor for stdin");

```

Comment:
Have you tried writing to `tracer` in this method and checking the behavior?  

At this point we're in the forked child process and I'm not sure what will happen if you attempt to use the `tracer` from the parent process.

Reply:
Works fine.  All handles are inherited in a child process so there doesn't seem to be any issues.

Reply:
> Works fine. All handles are inherited in a child process so there doesn't seem to be any issues.

Nice 😃  I am surprised that managed references are inherited, but I confirmed it works on my machine as well 👍